docs/v2/guides/gutenberg.md

```diff
@@ -11,112 +11,123 @@ Timber works with the Block Editor (also called Gutenberg) out of the box. If yo
-        'category' => 'formatting',
-        'icon' => 'admin-comments',
-        'keywords' => ['example'],
-    ]);
-}
+```
+.
+|--...
+|-- wp-content
+|   |-- themes 
+|       |-- your-theme # your theme directory
+|           |-- blocks # your blocks directory
+|               |-- my-block # your block
+|                   |-- block.json # your block settings 
+|                   |-- my-block.css # styles for your block
 ```
 
-Next, you you have to create your `render_callback()` function:
+#### Write the Block Settings file: block.json
 
-```php
-/**
- *  This is the callback that displays the block.
- *
- * @param   array  $block      The block settings and attributes.
- * @param   string $content    The block content (emtpy string).
- * @param   bool   $is_preview True during AJAX preview.
- */
-function my_acf_block_render_callback($block, $content = '', $is_preview = false)
+[Wordpress has a full example](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/) of the values you can add to your block.json. For our example, we'll follow [ACF's example](https://www.advancedcustomfields.com/resources/acf-blocks-key-concepts/#acf-blocks-and-blockjson) with the minimal fields plus the acf property. We will use renderCallback instead of renderTemplate so we can set a formula to render our blocks instead of rednering them one by one in our code. 
+
+```json
+//block.json
 {
-    $context = Timber::context([
-        // Store block values.
-        'block' => $block,
-        // Store field values.
-        'fields' => get_fields(),
-        // Store $is_preview value.
-        'is_preview' => $is_preview,
-    ]);
-
-    // Render the block.
-    Timber::render('block/example-block.twig', $context);
+    "name": "acf/my-block", // https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#name
+    "title": "My Block", // https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#title
+    "description": "Description for my block", // https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#description
+    "style": [ "file:./my-block.css" ], // https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#category,
+    
+    "category": "formatting", // https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#category
+    "icon": "format-aside", // https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#icon
+    "keywords": ["my", "block"], // https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#keywords
+    "acf": { 
+        "mode": "preview", // https://www.advancedcustomfields.com/resources/acf-blocks-key-concepts/#acf-blocks-and-blockjson
+        "renderCallback": "my_acf_block_render_callback" // the function that will render the block we'll add later on
+    } 
 }
```

Comment:
I would remove all the comments here so the creation is more easy. You could mention the separate keys underneath.

Reply:
that makes sense for copypasta. at my old gig, I would write comments for each value to map them for the other devs that would just use the json as a guide.